### PR TITLE
python3Packages.torchvision-bin: 0.9.1 -> 0.10.0

### PR DESCRIPTION
**Pull request by junjihashimoto**
###### Motivation for this change

python3Packages.pytorch-bin(1.9) needs torchvision-0.10.0, so we have to upgrade torchvision-bin to 0.10.0.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by junjihashimoto** [comment]
@danieldk 
This update works in this repo.
https://github.com/junjihashimoto/torchvision-fastercnn

File: pkgs/development/python-modules/torchvision/binary-hashes.nix [hunk 1]
```diff
@@ -3,20 +3,41 @@
...
   };
-  x86_64-linux-39 = {
-    name = "torchvision-${version}-cp39-cp39-linux_x86_64.whl";
-    url = "https://download.pytorch.org/whl/cu111/torchvision-${version}%2Bcu111-cp39-cp39-linux_x86_64.whl";
-    hash = "sha256-pzR7TBE+WcAmozskoeOVBuMkGJf9tvsaXsUkTcu86N8=";
+  "0.10.0" = {
+    x86_64-linux-37 = {
+      name = "torchvision-0.10.0-cp37-cp37m-linux_x86_64.whl";
+      url = "https://download.pytorch.org/whl/cu111/torchvision-0.10.0%2Bcu111-cp37-cp37m-linux_x86_64.whl";
+      sha256 = "1kh6zrid4jv2qh1fpgnghsy2p0k5dc9pmk90484bdziskn313j68";
```
Comment (samuela) [inline pkgs/development/python-modules/torchvision/binary-hashes.nix:30]: It would be nice to be consistent between `hash`/`sha256`. I believe that `hash` is preferred over `sha256`.

**Review by samuela — COMMENTED** [review]
Is there a reason to keep around the 0.9.1 version?

File: pkgs/development/python-modules/torchvision/binary-hashes.nix (hunk 1 above)
Comment (junjihashimoto) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix:30]: pytorch has lts(1.8.x) and latest one(1.9.x). 
Enterprise application may use lts-version.
General developers use  latest one.
0.9.1 version is for pytorch-1.8.x.

File: pkgs/development/python-modules/torchvision/prefetch.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+rm -f $hashfile
+echo "  \"$version\" = {" >> $hashfile
+
+for url_and_key in "${url_and_key_list[@]}"; do
+  key=$(echo "$url_and_key" | cut -d' ' -f1)
+  url=$(echo "$url_and_key" | cut -d' ' -f2)
+  name=$(echo "$url_and_key" | cut -d' ' -f3)
+
+  echo "prefetching ${url}..."
+  hash=$(nix hash to-base64 --type sha256 `nix-prefetch-url "$url" --name "$name"`)
```
Comment (samuela) [inline pkgs/development/python-modules/torchvision/prefetch.sh:23]: There's also `nix to-sri`, but as long as it works it seems fine to me either way!

**Review by samuela — APPROVED** [review]

File: pkgs/development/python-modules/torchvision/binary-hashes.nix (hunk 1 above)
Comment (junjihashimoto) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix]: @samuela 
Thank you for your reviewing!
Changed `sha256` to `hash`.

File: pkgs/development/python-modules/torchvision/bin.nix [hunk 3]
```diff
@@ -9,13 +9,13 @@
 , pillow
 , python
 , pytorch-bin
+, version ? "0.10.0"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torchvision/bin.nix:12]: Please revert this change.

File: pkgs/development/python-modules/torchvision/prefetch.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/torchvision/prefetch.sh:1]: Please use a nix-shell with all required tools.

File: pkgs/development/python-modules/torchvision/bin.nix (hunk 3 above)
Comment (junjihashimoto) [reply pkgs/development/python-modules/torchvision/bin.nix:12]: I can revert it, but it doesn't make sense to add version to `import ./binary-hashes.nix version`.

Comment (junjihashimoto) [reply pkgs/development/python-modules/torchvision/bin.nix:12]: @SuperSandro2000 
If we want torchvision-bin to support both lts and the latest, should I make torchvision-lts-bin and pytorch-lts-bin?

File: pkgs/development/python-modules/torchvision/binary-hashes.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix]: I think we should only keep the latest one. The general policy for Python packages in nixpkgs is that we do not ship multiple versions. Also, as long as there are no packages that depend on older versions, there is no real need. If someone needs the LTS version, they can always make their own copy of the derivation or override it.

Comment (samuela) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix]: If I'm understanding @junjihashimoto correctly, I think the issue is that we currently have a pytorch LTS version packaged in nixpkgs with some users, and for those users torchvision-bin must be packaged at 0.9.1.

Comment (danieldk) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix]: We have PyTorch 1.9.0:

https://github.com/NixOS/nixpkgs/blob/a3bf844c4781244c989c3367192c8b62eff50431/pkgs/development/python-modules/pytorch/bin.nix#L21
https://github.com/NixOS/nixpkgs/blob/a3bf844c4781244c989c3367192c8b62eff50431/pkgs/development/python-modules/pytorch/default.nix#L120

Comment (samuela) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix]: Oh well in that case no point keeping around the old version

Comment (junjihashimoto) [reply pkgs/development/python-modules/torchvision/binary-hashes.nix]: For now, I revert it. It looks better.

I understand that it is different from the overall python-package-management.
On the other hand, we continue to support multiple runtime versions not latest one only.

**Comment by junjihashimoto** [comment]
I've removed the code of 0.9.1.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 135874` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python38Packages.torchvision-bin</li>
<li>python39Packages.torchvision-bin</li>
</ul>
</details>

**Comment by junjihashimoto** [comment]
Thanks everyone!
